README.md

```diff
@@ -67,8 +68,28 @@ class MyCharm(CharmBase):
         # Handle the event here.
 ```
 
-Every standard event in Juju may be observed that way, and you can also easily
-define your own events in your custom types.
+In the above example, we are telling the framework we want to observe the
+`self.on.start` event and that we want the `self.on_start` handler to respond
+to that event. Every standard event in Juju may be observed that way, and you
+can also easily define your own events in your custom types.
```

Comment:
Since you're improving this section, it might be worth expanding the "custom types" bit